### DISCUSSION TITLE
Assistance Required for Deploying DIGIT-2.9LTS Business Services on AWS Cluster

### DISCUSSION THREAD

**Discussion by PrasadAthani** [egovernments/Digit-Core#596] [DIGIT Platform]
Dear Team,

I am currently working on the deployment of DIGIT-2.9LTS on an AWS cluster, including Backbone, Core, Business, and Municipal services. Thus far, I have successfully deployed the Backbone and Core services, which are up and running. However, I am encountering an issue while deploying the Business services.

The issue appears to be related to the PostgreSQL version in use. I am currently using the following version of PostgreSQL:
docker.io/bitnami/postgresql:17.0.0-debian-12-r0, which results in a connection failure when attempting to migrate the database.

Here is the error message I am receiving:
```
Flyway 4.1.2 by Boxfuse
ERROR:
Unable to obtain Jdbc connection from DataSource (jdbc:postgresql://postgres-postgresql.backbone:5432/postgres) for user 'postgres': The authentication type 10 is not supported.
SQL State  : 08004
Error Code : 0
Message    : The authentication type 10 is not supported. Check that you have configured the pg_hba.conf file to include the client's IP address or subnet, and that it is properly set up.
Stream closed EOF for egov/billing-service-644f5cb4b6-sw8zx (db-migration)
```
I suspect the PostgreSQL version might be incompatible with Flyway or other components in this setup. Could someone please guide me on the appropriate PostgreSQL version to use, or assist in troubleshooting this issue? Any insights on the configuration of the pg_hba.conf file to resolve the connection issue would also be greatly appreciated.

Looking forward to your support.

**Comment by pradeepkumarcm-egov** [comment]
Please refer document https://core.digit.org/platform/releases/digit-2.9-lts/upgrade-guide-transitioning-digit-modules-to-spring-boot-version-3.2.2. You need to follow the same steps to upgrade business and municipal service which you are using.  

  
Upgrade the Flyway library to version 9.22.3 for compatibility with Postgres 15.  We supporting postgress db till version 15. Please explore if you are using latest version of postgres and modify relevant library version. 

Below is the code snippet:

       <dependency>
         <groupId>org.flywaydb</groupId>
         <artifactId>flyway-core</artifactId>
         <version>9.22.3</version>
      </dependency>

Upgrade the postgresql library to version 42.7.1:
       <dependency>
           <groupId>org.postgresql</groupId>
           <artifactId>postgresql</artifactId>
           <version>42.7.1</version>
        </dependency>
